File: arrays-Hashing/leetCode_217.java
import java.util.HashSet;

public class leetCode_217 {
    public boolean containsDuplicates(int[] nums){
        HashSet<Integer> seen = new HashSet<>();
        for (int i : nums){
            if (seen.contains(i)){
                return true;
            }
            seen.add(i);
        }
        return false;
    }
    public static void main(String[] args){
        leetCode_217 solution = new leetCode_217();
        int[] nums = {1, 2, 3, 4, 5, 1};
        boolean result = solution.containsDuplicates(nums);
        System.out.println("Contains duplicates: " + result);
    }
}

File: arrays-Hashing/leetCode_217.py
#Contains Duplicate

"""
Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.
Ex : nums = [1,2,3,1]
output : True
"""


class Solution:
    def containsDuplicate(self, nums):
        elements = set()
        for number in nums:
            if number not in elements:
                elements.add(number)
            else:
                return True
        return False

s = Solution()
print(s.containsDuplicate([1,2,3,1]))